runtime/complex.go

```diff
@@ -130,3 +131,14 @@ func complexCoerce(o *Object) (complex128, bool) {
 	}
 	return complex(floatO, 0.0), true
 }
+
+func complexHash(f *Frame, o *Object) (*Object, *BaseException) {
+	v := toComplexUnsafe(o)
+	hashreal := hashFloat(real(v.Value()))
```

Comment:
Style nit: I don't think you can reduce the number of temporaries without affecting readability too much.